### PR TITLE
dts: msm8926: add Support for Samsung Galaxy S5 Mini Duos (kmini3g)

### PR DESCRIPTION
**Pull request by vicenteicc2008**
this adds support for Samsung Galaxy S5 Mini Duos (kmini3g)

### CODE REVIEW

File: dts/msm8226/msm8926-samsung-kmini3g.dts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <0xC808FF01 2 0x10001>;
```
Comment (TravMurav) [inline dts/msm8226/msm8926-samsung-kmini3g.dts:10]: This already corresponds to msm8926-samsung-r02.dts

Please move your kmini3g node there.

File: dts/msm8226/msm8926-samsung-kmini3g.dts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	kmini3g {
+		model = "Samsung Galaxy S5 Mini Duos (SM-G800H)";
+		compatible = "samsung,kmini3g", "qcom,msm8926", "lk2nd,device";
+		lk2nd,match-bootloader = "G800H*";
+
+		lk2nd,keys =
+			<KEY_HOME       108 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+			<KEY_VOLUMEDOWN 107 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+			<KEY_VOLUMEUP   106 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
```
Comment (TravMurav) [inline dts/msm8226/msm8926-samsung-kmini3g.dts:20]: Have you checked that all of these buttons work correctly? I.e. that you can move up and down in the menu and that the home button allows you to enter lk2nd fastboot  like the volume button does?

File: README.md [hunk 3]
```diff
@@ -119,6 +119,7 @@ page on the EFIDroid wiki for an exact mapping of LK targets to SoCs.
 - Motorola Moto G 2013 - falcon
 - Samsung Galaxy Grand 2 - SM-G7102
 - Samsung Galaxy Tab 4 10.1 (2014) - SM-T530, SM-T535
+- Samsung Galaxy S5 Mini Duos - SM-G800H
```
Comment (TravMurav) [inline README.md:122]: I believe S goes before T in the alphabet :)

**Review by TravMurav — COMMENTED** [review]
Thanks for this PR, here are some comments.

File: dts/msm8226/msm8926-samsung-kmini3g.dts (hunk 1 above)
Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-kmini3g.dts:10]: How?

File: dts/msm8226/msm8926-samsung-kmini3g.dts (hunk 2 above)
Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-kmini3g.dts:20]: I don't tested because

**Comment by vicenteicc2008** [comment]
buttons working fine

File: dts/msm8226/msm8926-samsung-kmini3g.dts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+        qcom,msm-id =  <0x6E08FF01 7 0x20000>;
```
[the remark this answers is not in the document]
Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-kmini3g.dts]: i renamed the dts

[the remark this answers is not in the document]
Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-kmini3g.dts]: not showing the model

**Comment by vicenteicc2008** [comment]
i am having error (MODEL- unknown (FIXME!)) and does not recognize in my computer

File: dts/msm8226/msm8926-samsung-r04.dts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+        qcom,msm-id =  <0x6E08FF01 4 0x20000>;
+	kmini3g {
+		model = "[ATLANTIC-3G] Rev-03-MTP-v2";
```
Comment (TravMurav) [inline dts/msm8226/msm8926-samsung-r04.dts:12]: if this device is "atlantic3g" call the node and compatible this way.
Take the msm-id from the same file you took this name from.

please keep this "model" property containing the "marketing" name of the device though (what's written on the box)

Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-r04.dts]: i updated the model and called node and compatible

File: dts/msm8226/rules.mk [hunk 6]
```diff
@@ -10,4 +10,5 @@ DTBS += \
 	$(LOCAL_DIR)/msm8226-samsung-ms013g.dtb \
 	$(LOCAL_DIR)/msm8926-huawei-g6-l11-vb.dtb \
 	$(LOCAL_DIR)/msm8926-samsung-r02.dtb \
-	$(LOCAL_DIR)/msm8926-v2-720p-mtp.dtb
+	$(LOCAL_DIR)/msm8926-v2-720p-mtp.dtb \
+	$(LOCAL_DIR)/msm8926-samsung-r04.dtb
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8226/rules.mk:14]: How about `msm8228` instead?
https://desktop.firmware.mobi/device:346

File: dts/msm8226/msm8926-samsung-r04.dts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+        qcom,msm-id =  <0x6E08FF01 4 0x20000>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8226/msm8926-samsung-r04.dts:10]: https://github.com/msm8916-mainline/lk2nd/blob/experimental-tmp2/lk2nd/include/dt-bindings/arm/qcom%2Cids.h#L90

Where do you know about `0x6E` (110)?

Comment (vicenteicc2008) [reply dts/msm8226/msm8926-samsung-r04.dts]: i renamed dts

File: dts/msm8226/msm8228-samsung-r03.dts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+        qcom,msm-id = <221 0>;
+	qcom,board-id = <0xdf08ff01 3>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8226/msm8228-samsung-r03.dts:11]: https://github.com/MSM8226-Samsung/android_kernel_samsung_kmini3g/blob/cm-13.0/arch/arm/boot/dts/msm8226/msm8228-sec-atlantic3g-r03.dts#L35-L36

I guess older SoC doesn't seem to support `qcom,board-id`

File: dts/msm8226/rules.mk [hunk 9]
```diff
@@ -10,4 +10,5 @@ DTBS += \
 	$(LOCAL_DIR)/msm8226-samsung-ms013g.dtb \
 	$(LOCAL_DIR)/msm8926-huawei-g6-l11-vb.dtb \
 	$(LOCAL_DIR)/msm8926-samsung-r02.dtb \
-	$(LOCAL_DIR)/msm8926-v2-720p-mtp.dtb
+	$(LOCAL_DIR)/msm8926-v2-720p-mtp.dtb \
+	$(LOCAL_DIR)/msm8228-samsung-r03.dtb
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8226/rules.mk:14]:
```
	...
	$(LOCAL_DIR)/msm8226-*.dtb \
	$(LOCAL_DIR)/msm8228-*.dtb \
	$(LOCAL_DIR)/msm8926-*.dtb \
	...
```

**Pull request closed without merging**